src/Stratis.Bitcoin/Consensus/ChainedHeaderTree.cs

```diff
@@ -357,6 +363,12 @@ public List<int> PartialOrFullValidationFailed(ChainedHeader chainedHeader)
         {
             this.logger.LogTrace("({0}:'{1}')", nameof(chainedHeader), chainedHeader);
 
+            if (!this.chainedHeadersByHash.ContainsKey(chainedHeader.HashBlock))
+            {
+                this.logger.LogTrace("(-)[NOT_FOUND]");
+                return new List<int>();
```

Comment:
We could return null here.

Reply:
Yes we could but in the other method we return an empty list. And imho it's more logically correct to return the empty list because method claim to return list of peers that should be banned. If it's empty- noone should be banned. 

Also since it's always not null the caller's logic will be cleaner.

Reply:
OK sure because this is a public method I accept this argument, if it was private I will probably got for null.